controllers/operator/openstack_controller.go

```diff
@@ -372,6 +380,146 @@ func (r *OpenStackReconciler) renderAndApply(
 	return nil
 }
 
+func isServiceOperatorResource(name string) bool {
+	//NOTE: test-operator was deployed as a independant package so it may or may not be installed
+	//NOTE: depending on how watcher-operator is released for FR2 and then in FR3 it may need to be
+	// added into this list in the future
+	serviceOperatorNames := []string{"barbican", "cinder", "designate", "glance", "heat", "horizon", "infra",
+		"ironic", "keystone", "manila", "mariadb", "neutron", "nova", "octavia", "openstack-baremetal", "ovn",
+		"placement", "rabbitmq-cluster", "swift", "telemetry", "test"}
+
+	for _, item := range serviceOperatorNames {
+		if strings.Index(name, item) == 0 {
+			return true
+		}
+	}
+	return false
+}
+
+// cleanupObsoleteResources - deletes CSVs and subscriptions
+func (r *OpenStackReconciler) cleanupObsoleteResources(ctx context.Context, instance *operatorv1beta1.OpenStack) error {
+	Log := r.GetLogger(ctx)
+
+	csvGVR := schema.GroupVersionResource{
+		Group:    "operators.coreos.com",
+		Version:  "v1alpha1",
+		Resource: "clusterserviceversions",
+	}
+
+	subscriptionGVR := schema.GroupVersionResource{
+		Group:    "operators.coreos.com",
+		Version:  "v1alpha1",
+		Resource: "subscriptions",
+	}
+
+	installPlanGVR := schema.GroupVersionResource{
+		Group:    "operators.coreos.com",
+		Version:  "v1alpha1",
+		Resource: "installplans",
+	}
+
+	csvList := &uns.UnstructuredList{}
+	csvList.SetGroupVersionKind(csvGVR.GroupVersion().WithKind("ClusterServiceVersion"))
+	err := r.Client.List(ctx, csvList, &client.ListOptions{Namespace: instance.Namespace})
+	if err != nil {
+		return err
+	}
+	for _, csv := range csvList.Items {
+		Log.Info("Found CSV", "name", csv.GetName())
+		if isServiceOperatorResource(csv.GetName()) {
+			err = r.Client.Delete(ctx, &csv)
+			if err != nil {
```

Comment:
Maybe check for "not found" error here in case the CSV was somehow deleted (in which case it can be ignored and we can just continue)?  Probably won't happen, but just a thought.

Reply:
sure, I can add a check for not found